### PR TITLE
 [MM-21901] Set trigger timeout for Interactive Dialogue dynamically using config instead of const value

### PR DESCRIPTION
**Pull request by Apahadi73**
#### Summary

 Set trigger timeout for Interactive Dialogue dynamically using config instead of const value to allow the client/user to decide the dialogue timeout issue instead of using a constant timeout value

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/21901
#### Related Pull Requests

- Has client changes ([[MM-21901] : Add Interactive Dialog Trigger Timeout config option in Integration Management setting](https://github.com/mattermost/mattermost-webapp/pull/12356))

#### Release Note

```release-note
 Set trigger timeout for Interactive Dialogue dynamically using config instead of const value
```

[MM-21901]: https://mattermost.atlassian.net/browse/MM-21901?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by neallred** [comment]
Thanks for these changes, @Apahadi73 ! Looks like the linter found another spot that needs changing because of the added argument: https://github.com/mattermost/mattermost-server/actions/runs/4394238185/jobs/7716391812 . After that is changed, this PR looks good to me!

File: model/integration_action.go [hunk 1]
```diff
@@ -280,7 +279,7 @@ func (r *PostActionIntegrationRequest) GenerateTriggerId(s crypto.Signer) (strin
 	return clientTriggerId, triggerId, nil
 }
 
-func DecodeAndVerifyTriggerId(triggerId string, s *ecdsa.PrivateKey) (string, string, *AppError) {
+func DecodeAndVerifyTriggerId(triggerId string, s *ecdsa.PrivateKey, InteractiveDialogTriggerTimeout int) (string, string, *AppError) {
```
Comment (mickmister) [inline model/integration_action.go:282]:
```suggestion
func DecodeAndVerifyTriggerId(triggerId string, s *ecdsa.PrivateKey, interactiveDialogTriggerTimeout int) (string, string, *AppError) {
```

File: model/integration_action.go [hunk 2]
```diff
@@ -327,8 +326,8 @@ func DecodeAndVerifyTriggerId(triggerId string, s *ecdsa.PrivateKey) (string, st
 	return clientTriggerId, userId, nil
 }
 
-func (r *OpenDialogRequest) DecodeAndVerifyTriggerId(s *ecdsa.PrivateKey) (string, string, *AppError) {
-	return DecodeAndVerifyTriggerId(r.TriggerId, s)
+func (r *OpenDialogRequest) DecodeAndVerifyTriggerId(s *ecdsa.PrivateKey, InteractiveDialogTriggerTimeout int) (string, string, *AppError) {
```
Comment (mickmister) [inline model/integration_action.go:329]:
```suggestion
func (r *OpenDialogRequest) DecodeAndVerifyTriggerId(s *ecdsa.PrivateKey, interactiveDialogTriggerTimeout int) (string, string, *AppError) {
```

**Review by mickmister — COMMENTED** [review]
Awesome work on this @Apahadi73!

I have just a few asks for have the names of function parameters to be in `camelCase` instead of `PascalCase`. Otherwise LGTM!

File: model/integration_action.go (hunk 1 above)
Comment (Apahadi73) [reply model/integration_action.go]: done👍

File: model/integration_action.go (hunk 2 above)
Comment (Apahadi73) [reply model/integration_action.go]: done👍

**Comment by Apahadi73** [comment]
> Thanks for these changes, @Apahadi73 ! Looks like the linter found another spot that needs changing because of the added argument: https://github.com/mattermost/mattermost-server/actions/runs/4394238185/jobs/7716391812 . After that is changed, this PR looks good to me!

done👍. Thanks for the suggestion.

**Comment by neallred** [comment]
I see one more lint: https://github.com/mattermost/mattermost-server/actions/runs/4411551122/jobs/7740268583 . After it's addressed I think this is PR ready.

**Comment by Apahadi73** [comment]
> I see one more lint: https://github.com/mattermost/mattermost-server/actions/runs/4411551122/jobs/7740268583 . After it's addressed I think this is PR ready.

resolved

**Comment by neallred** [comment]
Looks like there's still a mismatched call in model/integration_action_test.go:39:90: https://github.com/mattermost/mattermost-server/actions/runs/4421613148/jobs/7764585821

**Comment by Apahadi73** [comment]
> Looks like there's still a mismatched call in model/integration_action_test.go:39:90: https://github.com/mattermost/mattermost-server/actions/runs/4421613148/jobs/7764585821

done

**Comment by mickmister** [comment]
Hi @Apahadi73, we made some changes to our repository and moved around some files (coincidentally the first time we've done this in a long time). Are you open to resolving the merge conflicts on the PR?

**Comment by hanzei** [comment]
Replaced by https://github.com/mattermost/mattermost/pull/23805. Thanks for this contribution @Apahadi73

**Pull request closed without merging**